libfreerdp/core/rdp.c

```diff
@@ -373,6 +373,14 @@ BOOL rdp_read_header(rdpRdp* rdp, wStream* s, UINT16* length, UINT16* channelId)
 			else
 				rdp_set_error_info(rdp, ERRINFO_RPC_INITIATED_DISCONNECT);
 		}
+		else if ((rdp->errorInfo == ERRINFO_RPC_INITIATED_DISCONNECT) && (reason == MCS_Reason_user_requested))
```

Comment:
I'd map the `ERRINFO_RPC_INITIATED_DISCONNECT` code to a return code instead of setting a (unrelated) error info code.

Reply:
This function (`rdp_read_header()`) has a return type of BOOL, so little information can be passed through that.  Also, the `rdp_set_error_info()` call is the same as that used a few lines above for `reason == NCS_Reason_user_requested`, so it seems at least somewhat consistent.

However, I suspect I'm missing your point.  Can you elaborate?

Reply:
@r-barnett but the error code set will not be overridden. That can be checked and converted to a proper exit code.

Reply:
@akallabeth Yes, that's what happens with `ERRINFO_LOGOFF_BY_USER`, but...you already understand that, so I'm still missing something.

Are you suggesting the use of a different error code or maybe the creation of a new error code?